src/components/generics/vacation-test-forms/myVacationRequests.tsx

```diff
@@ -1,287 +1,349 @@
-                    <>
-                      <TableRow>
-                        <TableCell>
-                          <Box className={ classes.datePickers }>
-                            <DateRangePicker
-                              scope={ FilterScopes.DATE }
-                              dateFormat={ dateFormat }
-                              selectedStartDate={ selectedVacationStartDate }
-                              selectedEndDate={ selectedVacationEndDate }
-                              datePickerView={ datePickerView }
-                              minStartDate={ new Date() }
-                              minEndDate={ selectedVacationStartDate }
-                              onStartDateChange={ handleVacationStartDateChange }
-                              onEndDateChange={ handleVacationEndDateChange }
-                              onStartWeekChange={() => {
-                                throw new Error("Function not implemented.");
-                              } }
-                              onEndWeekChange={() => {
-                                throw new Error("Function not implemented.");
-                              } }
-                            />
-                          </Box>
-                        </TableCell>
-                        <TableCell>
-                          { renderVacationType() }
-                        </TableCell>
-                        <TableCell>
-                          { renderVacationDaysSpent() }
-                          { renderVacationCommentBox() }
-                          <Box display="flex" justifyContent="center">
-                            { renderVacationApplyButton() }
-                          </Box>
-                        </TableCell>
-                        <TableCell>
-                          <IconButton
-                            onClick={ deleteRequest }
-                            aria-label="delete"
-                            className={ classes.deleteButton }
-                            size="large"
-                          >
-                            <DeleteIcon fontSize="medium"/>
-                          </IconButton>
-                        </TableCell>
-                      </TableRow>
-                    </>
-                  </Collapse>
-                </TableCell>
-              </>
-            ))}
-          </TableBody>
-        </Table>
-      </TableContainer>
+                      { handleRequestStatus(request.hrManagerStatus) }
+                    </StyledTableCell>
+                    <TableCell>
+                      <IconButton
+                        aria-label="expand row"
+                        size="small"
+                        onClick={() => {
+                          const newOpenRows = [...openDetails];
+                          newOpenRows[index] = !newOpenRows[index];
+                          setOpenDetails(newOpenRows);
+                        }}
```

Comment:
Moving this logic into an appropriately named function could improve code readability